game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/ProTechAi.java

```diff
@@ -48,9 +48,10 @@ static void tech(final ITechDelegate techDelegate, final GameData data, final Ga
     final float enemyStrength = getStrengthOfPotentialAttackers(myCapitol, data, player);
     float myStrength = getMyStrength(data, player, myCapitol);
     final boolean capDanger = myStrength < (enemyStrength * 1.25F + 3.0F);
-    final Resource pus = data.getResourceList().getResource(Constants.PUS);
+    final Resource pus = data.getResourceList().getResource(Constants.PUS).orElse(null);
```

Comment:
@BBlaza Same, either the method assumes the value must be available, then new method `getResourceOrThrow?` or a default value should be used, e.g., `orElse(0)`.